lib/pf/api.pm

```diff
@@ -572,6 +572,25 @@ sub register_node : Public {
     return;
 }
 
+=head2 register_node_ip
+
+Register a node by ip address
+
+=cut
+
+sub register_node_ip : Public {
+    my ($class, %postdata )  = @_;
+    my @require = qw(ip pid);
+    my @found = grep {exists $postdata{$_}} @require;
+    return unless pf::util::validate_argv(\@require,  \@found);
+
+    my $mac = pf::iplog::ip2mac($postdata{'ip'});
+    die "Cannot find host with IP address $postdata{'ip'}" unless $mac;
+
+    pf::node::node_register($mac, $postdata{'pid'}, %postdata);
+    return;
```

Comment:
> That's ok, do you mean we can change the API register_node's return value as well ?

Might as well while you are at it.

> BTW, do you know why the pf::node::node_register returns (1|0) instead of boolean directly ?

Booleans don't exist in perl (at least the version we use) $TRUE and $FALSE are constants pointing to 0 and 1. We just haven't rewriten all of the 0s and 1s to those yet